src/m365/spo/commands/tenant/tenant-applicationcustomizer-add.ts

```diff
@@ -0,0 +1,200 @@
+      const solutionId = await this.getSolutionIdFromComponentManifestList(appCatalogUrl, args.options.clientSideComponentId, logger);
+      const solution = await this.getSolutionFromAppCatalog(appCatalogUrl, solutionId, logger);
+
+      if (!solution.ContainsTenantWideExtension) {
+        throw `The solution does not contain an extension that can be deployed to all sites. Make sure that you've entered the correct component Id.`;
+      }
+      else if (!solution.SkipFeatureDeployment) {
+        throw 'The solution has not been deployed to all sites. Make sure to deploy this solution to all sites.';
+      }
+      await this.addTenantWideExtension(appCatalogUrl, args.options, logger);
+    }
+    catch (err: any) {
+      this.handleRejectedODataJsonPromise(err);
+    }
+  }
+
+  private async getAppCatalogUrl(logger: Logger): Promise<string> {
+    const spoTenantAppCatalogUrlGetCommandOutput: CommandOutput = await Cli.executeCommandWithOutput(spoTenantAppCatalogUrlGetCommand as Command, { options: { output: 'text', _: [] } });
+    if (this.verbose) {
+      logger.logToStderr(spoTenantAppCatalogUrlGetCommandOutput.stderr);
+    }
+    const appCatalogUrl: string | undefined = spoTenantAppCatalogUrlGetCommandOutput.stdout;
+    if (!appCatalogUrl) {
+      throw 'Cannot add tenant-wide application customizer as app catalog cannot be found';
+    }
+    if (this.verbose) {
+      logger.logToStderr(`Got tenant app catalog url: ${appCatalogUrl}`);
+    }
+    return appCatalogUrl;
+  }
+
+  private async getSolutionIdFromComponentManifestList(appCatalogUrl: string, clientSideComponentId: string, logger: Logger): Promise<string> {
+    if (this.verbose) {
+      logger.logToStderr('Retrieving component manifest item from the ComponentManifests list on the app catalog site so that we get the solution id');
+    }
+
+    const camlQuery = `<View><ViewFields><FieldRef Name='ClientComponentId'></FieldRef><FieldRef Name='SolutionId'></FieldRef><FieldRef Name='ClientComponentManifest'></FieldRef></ViewFields><Query><Where><Eq><FieldRef Name='ClientComponentId' /><Value Type='Guid'>${clientSideComponentId}</Value></Eq></Where></Query></View>`;
+    const commandOptions: spoListItemListCommandOptions = {
+      webUrl: appCatalogUrl,
+      listUrl: `${urlUtil.getServerRelativeSiteUrl(appCatalogUrl)}/Lists/ComponentManifests`,
+      camlQuery: camlQuery,
+      verbose: this.verbose,
+      debug: this.debug,
+      output: 'json'
+    };
+    const output = await Cli.executeCommandWithOutput(spoListItemListCommand as Command, { options: { ...commandOptions, _: [] } });
+    if (this.verbose) {
+      logger.logToStderr(output.stderr);
+    }
+    const outputParsed = JSON.parse(output.stdout);
+    if (outputParsed.length === 0) {
+      throw 'No component found with the specified clientSideComponentId found in the component manifest list. Make sure that the application is added to the application catalog';
+    }
+    logger.log(outputParsed);
```

Comment:
Is this line supposed to be here? I think not :-)

Reply:
Woopsie! 😄